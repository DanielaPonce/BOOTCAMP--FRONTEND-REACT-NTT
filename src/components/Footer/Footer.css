/* Footer */
footer {
	background-color: var(--white);
	color: var(--text-color);
	padding: 1.5rem;
	text-align: center;
}

footer .social-icons {
	margin: 1rem 0;
}

.social-icons img {
	cursor: pointer;
}

.social-icons img:hover {
	filter: invert(50%) sepia(90%) saturate(1000%) hue-rotate(350deg)
		brightness(95%) contrast(90%);
}

footer .social-icons a:hover {
	color: var(--secondary-color);
}
