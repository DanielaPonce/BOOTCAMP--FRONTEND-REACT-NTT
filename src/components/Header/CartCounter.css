.cart-icon {
	cursor: pointer;
	position: relative;
	width: 30px;
	height: 30px;
}

.cart-icon .cart-count {
	position: absolute;
	top: -8px;
	right: -8px;
	background-color: var(--error-color);
	color: var(--secondary-color);
	border-radius: 50%;
	padding: 0.2rem 0.5rem;
	font-size: 0.8rem;
}

@media only screen and (max-width: 739px) {
	.cart-icon {
		position: absolute;
		top: 1.5rem;
		right: 2rem;
	}

}