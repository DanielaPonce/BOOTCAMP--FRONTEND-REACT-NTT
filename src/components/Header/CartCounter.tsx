import { FC } from 'react';
import { useCart } from '../../hooks/useCart';
import { Link } from 'react-router';
import { RoutesConstants } from '../../utils/routes';

const CartCounter: FC = () => {
	const { itemCount } = useCart();

	return (
		<div className="cart-icon">
			<Link to={RoutesConstants.summary}>
				<img src="/src/assets/icons/cart-icon.svg" alt="cart icon" />
			</Link>

			{itemCount > 0 ? (
				<span id="cardCounter" className="cart-count">
					{itemCount}
				</span>
			) : (
				<></>
			)}
		</div>
	);
};

export default CartCounter;
