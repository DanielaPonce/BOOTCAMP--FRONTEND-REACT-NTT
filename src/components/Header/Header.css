.header-container {
	background-color: var(--primary-color);
	position: relative;
	height: 12vh;
}

.header-container header {
	position: absolute;
	top: 5vh;
	width: 90vw;
	left: 50%; 

	transform: translate(-50%, 7%); 
	box-shadow: 5px 5px 15px rgba(0, 0, 0, 22%);
	background-color: var(--white);
	border-radius: 25px;
	color: var(--text-color);
	padding: 2em 24px;

	display: flex;
	justify-content: space-between;
	align-items: center;
}

header .search-container {
	display: flex;
	justify-content: center;
	align-items: center;
	gap: 8px;
}

.logo {
	display: flex;
	align-items: center;
	gap: 0.5rem;
	text-decoration: none;
	color: var(--text-color);
	cursor: pointer;
}

.logo h1:hover {
	color: var(--hover-button-color);
}


@media only screen and (min-width: 740px) and (max-width: 991px) {
	.logo h1 {
		font-size: 24px;
	}
	
}

@media only screen and (max-width: 739px) {
	.header-container header {
		flex-direction: column;
		top: 1vh;
		padding: 16px;
		gap: 16px;
		width: 85vw;
	}

	.logo h1 {
		margin: 0;
		font-size: 24px;
	}
}
