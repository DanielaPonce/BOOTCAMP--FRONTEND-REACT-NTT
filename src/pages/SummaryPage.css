.summary-container {
	margin: 6rem 15rem;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	gap: 3rem;
}

@media only screen and (min-width: 740px) and (max-width: 1024px) {
	.summary-container {
		margin: 6rem 2rem;
	}
}
@media only screen and (max-width: 739px) {
	.summary-container {
		margin: 3rem 2rem;
	}
}
