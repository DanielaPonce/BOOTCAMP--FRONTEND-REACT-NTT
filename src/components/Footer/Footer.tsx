import './Footer.css'

const Footer = () => {
	return (
		<footer>
			<p>Síguenos en nuestras redes sociales</p>
			<div className="social-icons">
				<img src="/src//assets/icons/fb-icon.svg" alt="facebook icon" />
				<img
					src="/src/assets/icons/instagram-icon.svg"
					alt="instagram icon"
				/>
				<img
					src="/src/assets/icons/twitter-icon.svg"
					alt="twitter icon"
				/>
			</div>
			<p>© 2024 My Market. Todos los derechos reservados.</p>
			<p>
				<a href="#">Política de privacidad</a> |
				<a href="#">Términos de uso</a> | <a href="#">Contacto</a>
			</p>
		</footer>
	);
};

export default Footer;
