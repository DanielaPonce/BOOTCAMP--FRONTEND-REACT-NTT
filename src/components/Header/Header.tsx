import { FC } from 'react';
import { Link, useLocation } from 'react-router';
import { useCategories } from '../../hooks/useCategories';
import { StaticTexts } from '../../utils/staticTexts';
import CartCounter from './CartCounter';
import CategoryDropdown from './CategoryDropdown';
import SearchBar from './SearchBar';
import { RoutesConstants } from '../../utils/routes';
import { Images } from '../../utils/images';
import './Header.css';
import { useAuth } from '../../context/AuthContext';
import Button from '../Button/Button';

type HeaderProps = {
	onSearch?: (query: string) => void;
	onFilterCategory?: (category: string) => void;
};

const Header: FC<HeaderProps> = ({ onSearch, onFilterCategory }) => {
	const { user, logout } = useAuth();
	const location = useLocation();

	const { categories } = useCategories();

	const hideSearchAndDropdown = location.pathname === '/summary';

	return (
		<div className="header-container">
			<header>
				<div className="left-side">
					<Link to={RoutesConstants.Home} className="logo">
						<img
							src={Images.Logo}
							alt="Logo"
							width={45}
							height={45}
						/>
						<h1>{StaticTexts.AppTitle}</h1>
					</Link>
					{user && <span>Bienvenido: {user}</span>}
				</div>

				<div className="search-container">
					{hideSearchAndDropdown ? (
						<></>
					) : (
						<>
							<SearchBar onInputSearch={onSearch!} />
							<CategoryDropdown
								categories={categories}
								onSelectCategory={onFilterCategory!}
							/>
						</>
					)}
					<CartCounter />
					<Button
						type="button"
						className="logout-button"
						onClick={logout}
					>
						Cerrar Sesi√≥n
					</Button>
				</div>
			</header>
		</div>
	);
};

export default Header;
