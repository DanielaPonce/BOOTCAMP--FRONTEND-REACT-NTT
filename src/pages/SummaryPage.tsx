import CartSummary from '../components/Cart/CartSummary';
import CartTable from '../components/Cart/CartTable';
import Footer from '../components/Footer/Footer';
import Header from '../components/Header/Header';
import ShippingForm from '../components/ShippingForm/ShippingForm';

const SummaryPage = () => {
	return (
		<>
			<Header />
			<div className="summary-container">
				<CartTable />
				<CartSummary />
				<ShippingForm />
			</div>
			<Footer />
		</>
	);
};

export default SummaryPage;
