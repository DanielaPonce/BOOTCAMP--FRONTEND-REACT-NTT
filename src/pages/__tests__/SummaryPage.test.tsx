import { jest } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import Footer from '../../components/Footer/Footer';
import Header from '../../components/Header/Header';
import SummaryPage from '../SummaryPage';

jest.mock('../../components/Header/Header', () => ({
	__esModule: true,
	default: jest.fn(() => <header>Header</header>)
}));

jest.mock('../../components/Footer/Footer', () => ({
	__esModule: true,
	default: jest.fn(() => <footer>Footer</footer>)
}));

jest.mock('../../components/Cart/CartTable', () => ({
	__esModule: true,
	default: jest.fn(() => <div>CartTable</div>)
}));

jest.mock('../../components/Cart/CartSummary', () => ({
	__esModule: true,
	default: jest.fn(() => <div>CartSummary</div>)
}));

jest.mock('../../components/ShippingForm/ShippingForm', () => ({
	__esModule: true,
	default: jest.fn(() => <form>ShippingForm</form>)
}));

describe('SummaryPage Component', () => {
	beforeEach(() => {
		jest.clearAllMocks();
	});

	test('renders Header, Footer, and main components', () => {
		render(<SummaryPage />);

		expect(Header).toHaveBeenCalledTimes(1);
		expect(Footer).toHaveBeenCalledTimes(1);

		expect(screen.getByText('CartTable')).toBeInTheDocument();
		expect(screen.getByText('CartSummary')).toBeInTheDocument();
		expect(screen.getByText('ShippingForm')).toBeInTheDocument();
	});

	test('matches snapshot', () => {
		const { asFragment } = render(<SummaryPage />);
		expect(asFragment()).toMatchSnapshot();
	});
});
