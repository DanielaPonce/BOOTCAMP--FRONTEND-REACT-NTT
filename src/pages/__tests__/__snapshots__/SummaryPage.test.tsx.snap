// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryPage Component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    Header
  </header>
  <div
    class="summary-container"
  >
    <div>
      CartTable
    </div>
    <div>
      CartSummary
    </div>
    <form>
      ShippingForm
    </form>
  </div>
  <footer>
    Footer
  </footer>
</DocumentFragment>
`;
